soroban-env-host/src/cost_runner/cost_types/host_mem_cpy.rs

```diff
@@ -0,0 +1,18 @@
+use crate::{
+    budget::{AsBudget, CostType},
+    cost_runner::CostRunner,
+    host::metered_clone::MeteredClone,
+};
+
+pub struct HostMemCpyRun;
+
+impl CostRunner for HostMemCpyRun {
+    const COST_TYPE: CostType = CostType::HostMemCpy;
+    type SampleType = ([u8; 32], u64);
+
+    fn run_iter(host: &crate::Host, _iter: u64, sample: Self::SampleType) {
+        for _ in 0..sample.1 {
+            sample.0.metered_clone(host.as_budget()).unwrap();
```

Comment:
I think this is a little misleading in terms of measuring byte-copies of varying size.

1. It's only copying 32 bytes per iteration, which means it's mixing in N/32 calls to budget-charging.
2. It's also mixing in the cost of the loop, getting the budget from the host, calling metered clone, unwrapping the result..
3. It's not 100% clear to me that it's actually going to _do_ much of a byte-copy here since the result isn't used, it's possible the optimizer will eliminate the very part you're trying to measure

I would approach this by making the SampleType a pair of Vec<u8>, both the same size, and both held in an Rc<RefCell<Vec<u8>>> to ensure the optimizer doesn't eliminate them, then making the size of the vectors just vary directly with the input request, and doing a single metered call to a byte-copy loop (eg. std::slice::copy_from_slice) one vector to another. If there's no "metered byte-copy" call on the host currently to accumulate the budget input for this, add one.

Reply:
I have fixed this and adopted the "recycling the inputs back to the caller" approach.

Reply:
Great, looks good!